moveit_simple/src/moveit_simple.cpp

```diff
@@ -358,62 +353,72 @@ void Robot::execute(const std::string traj_name)
 bool Robot::toJointTrajectory(const std::string traj_name,
                        std::vector<trajectory_msgs::JointTrajectoryPoint> & points)
 {
-  const Trajectory & traj = traj_map_[traj_name];
-  const std::vector< std::string> & traj_type = traj_info_map_[traj_name].first;
-  const std::vector<unsigned int> num_steps = traj_info_map_[traj_name].second;
+  const TrajectoryInfo & traj_info = traj_info_map_[traj_name];
 
   // The first point in any trajectory is the current pose
   std::vector<double> current_joint_position;
   robot_state_->copyJointGroupPositions(joint_group_->getName(), current_joint_position);
   points.push_back(toJointTrajPtMsg(current_joint_position, 0.0));
 
-  for(size_t i = 0; i < traj.size(); ++i)
+  for(size_t i = 0; i < traj_info.size(); ++i)
   {
-    const std::unique_ptr<TrajectoryPoint> & traj_point = traj[i];
-    if (traj_type[i] != "CARTESIAN")
+    const std::unique_ptr<TrajectoryPoint> & traj_point = traj_info[i].point;
+    if (traj_info[i].type == JOINT)
```

Comment:
It's better practice to use a `switch` statement with ENUMs.  Not critical, fix if you want to.